activesupport/lib/active_support/parameter_filter.rb

```diff
@@ -110,7 +110,14 @@ def value_for_key(key, value, parents = [], original_params = nil)
         elsif value.is_a?(Hash)
           value = call(value, parents, original_params)
         elsif value.is_a?(Array)
-          value = value.map { |v| v.is_a?(Hash) ? call(v, parents, original_params) : v }
+          # If we don't pop of the current parent it will be duplicated as we
+          # process each array value.
+          parents.pop if deep_regexps
```

Comment:
This feels kind of gross, but it also felt weird for the same parent to get pushed onto the stack twice when mapping over the array of values.